### ISSUE TITLE
no-unused-modules does not work with `export class`

### ISSUE THREAD

**Issue by ehmicky** [benmosher/eslint-plugin-import#1368]
`index.js`:

```js
import { Example } from './foo.js'
export { Example }
```

`foo.js`:

```js
export class Example {}
```

`.eslintrc.yml`:

```yml
parserOptions:
  ecmaVersion: 2019
  sourceType: module
plugins: [import]
rules:
  import/no-unused-modules: [2, {missingExports: true}]
```

Then:

```bash
$ eslint
/home/user/example/foo.js
  1:1  error  No exports found                                 import/no-unused-modules

✖ 1 problem (1 error, 0 warnings)
```

However the following works:

```js
class Example {}
export { Example }
```

`eslint`: `5.16.0`
`eslint-plugin-import`: `2.17.3`
`node`: `12.3.1`
OS: Ubuntu `19.04`

**Comment by ehmicky** [comment]
Thanks for the fix! :tada:

**Comment by oleksandrhinko** [comment]
I've got the same issue.

`{
  "env": {
    "node": true,
    "es6": true
  },
  "extends": ["eslint:recommended"],
  "plugins": ["import"],
  "rules": {
    "eqeqeq": "error",
    "no-unused-expressions": "error",
    "no-unused-vars": ["warn", { "vars": "all", "args": "none" }],
    "import/no-unused-modules": [
      1,
      { "unusedExports": true, "missingExports": true }
    ]
  }
}`

**Comment by nathanielcook** [comment]
It seems like this same problem exists for `export module Example`, or am I misinterpreting?

**Comment by ljharb** [comment]
@nathanielcook `export module` is not valid syntax.

**Comment by nathanielcook** [comment]
@ljharb It is in TypeScript; sorry for not mentioning that.

**Comment by ljharb** [comment]
In that case, please feel free to file a PR to add it :-)

**Comment by nathanielcook** [comment]
By the way, I see now that the problem also exists for `export enum` and `export interface`.
